plugins/pdi-pur-plugin/src/org/pentaho/di/repository/pur/LazyUnifiedRepositoryDirectory.java

```diff
@@ -129,12 +130,10 @@ public LazyUnifiedRepositoryDirectory( RepositoryFile self, RepositoryDirectoryI
         UnifiedRepositoryLockService lockService =
             (UnifiedRepositoryLockService) registry.getService( ILockService.class );
 
-        RepositoryRequest repositoryRequest = new RepositoryRequest();
-        repositoryRequest.setShowHidden( true );
-        repositoryRequest.setTypes( RepositoryRequest.FILES_TYPE_FILTER.FILES );
-        repositoryRequest.setPath( this.self.getId().toString() );
-        List<RepositoryFile> children = repository.getChildren( repositoryRequest );
-        for ( RepositoryFile child : children ) {
+        RepositoryFileTree tree = repository.getTree( new RepositoryRequest( this.self.getPath(), true, 1, null ) );
```

Comment:
The version enablement is only available on the RepositoryFileTree class which isn't good. Rather than plumbing that through RepositoryFile, I'm just switching back to a getTree() call with a depth of 1.